lib/kafka_rest_client.js

```diff
@@ -100,16 +103,29 @@ KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry
 KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRefresh() {
     var self = this;
     if (self.proxyRefreshTime > 0) {
-        // console.log('Trying to schedule topic refresh job every: ' +
-        // self.proxyRefreshTime / 1000 + ' seconds.');
         /* eslint-disable no-undef,block-scoped-var */
         self.refreshIntervalId = setInterval(function refreshTopics() {
+            self.scheduleConnect(0);
+        }, self.proxyRefreshTime);
+        /* eslint-enable no-undef,block-scoped-var */
+    }
+};
+
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleConnect = function scheduleConnect(retry) {
+    var self = this;
+    if (retry < connectWaitTimeMs.length) {
```

Comment:
If there is a 3 second hiccup we are down for 30 minutes ?

Change the retries to something more forgiving;

```js
var connectWaitTimeMs = [0, 100, 3000, 10000, 30000, 60000]; 
```